app/models/spree/gateway/pay_pal_express.rb

```diff
@@ -33,6 +33,62 @@ def auto_capture?
     def method_type
       'paypal'
     end
+    
+    # This is rather hackish, required for payment/processing handle_response code.
+    def empty_success
+      Class.new do
+        def success?; true; end
+        def authorization; nil; end
+      end.new
+    end
+
+    #this is essentially the refund method below except the creation of the refund transaction
+    #as that will count doubly towards the "credited" amount
+    def void(response_code, gateway_options={})
+      payment = Spree::Payment.find_by_response_code(response_code)
+      amount = payment.credit_allowed
+
+      #in case a partially refunded payment gets cancelled/voided, we don't want to act on the refunded payments
+      if amount.to_f > 0
+
+        # Process the refund
+        refund_type = payment.amount == amount.to_f ? "Full" : "Partial"
+
+        refund_transaction = provider.build_refund_transaction({
+          :TransactionID => payment.source.transaction_id,
+          :RefundType => refund_type,
+          :Amount => {
+            :currencyID => payment.currency,
+            :value => amount },
+          :RefundSource => "any" })
+        
+        refund_transaction_response = provider.refund_transaction(refund_transaction)
+      
+        if refund_transaction_response.success?
+          payment.source.update_attributes({
+            :refunded_at => Time.now,
+            :refund_transaction_id => refund_transaction_response.RefundTransactionID,
+            :state => "refunded",
+            :refund_type => refund_type
+          })
+          empty_success
+        else
+          class << refund_transaction_response
+            def to_s
+              errors.map(&:long_message).join(" ")
+            end
+          end
+          refund_transaction_response
+        end
+      end
+
+      empty_success
```

Comment:
Likely not, but frankly I haven't touched this code since so I'm not even sure either way. You're going to have to test it out and see what happens.